pyproject.toml

```diff
@@ -46,19 +50,32 @@ classifiers = [
   "Topic :: Scientific/Engineering",
   "Topic :: Software Development",
 ]
+# core dependencies of sktime
+# this set should be kept minimal!
 dependencies = [
-  "numpy<1.27,>=1.21",
-  "packaging",
-  "pandas<2.2.0,>=1.1",
-  "scikit-base<0.7.0",
-  "scikit-learn<1.4.0,>=0.24",
-  "scipy<2.0.0,>=1.2",
+  "numpy<1.27,>=1.21",  # required for framework layer and base class logic
+  "packaging",  # for estimator specific dependency parsing
+  "pandas<2.2.0,>=1.1",  # pandas is the main in-memory data container
+  "scikit-base<0.7.0",  # base module for sklearn compatible base API
```

Comment:
Same as above, suggesting lower bound.

Reply:
this PR scope is only adding comments and possibly changing the order, but not the comment of `pyproject.toml`.